src/ssl/support.cc

```diff
@@ -748,6 +786,8 @@ Ssl::Initialize(void)
 
     SQUID_OPENSSL_init_ssl();
 
+    CRYPTO_set_mem_functions(CryptoMalloc, CryptoRealloc, CryptoFree);
```

Comment:
> CRYPTO_set_mem_functions() returns 1 on success or 0 on failure (almost always because allocations have already happened).

Please handle errors with a level-1 ERROR message. I do not know whether we can print some OpenSSL-provided error code or reason here. You may be able to tell from CRYPTO_set_mem_functions() implementation if that function sets something like that for caller retrieval.

Reply:
Done.

>  CRYPTO_set_mem_functions() implementation

This function just checks a flag that the standard CRYPTO_malloc() function sets upon its first call (and returns 0 if yes).